### DISCUSSION TITLE
Performance issue on macOS Ventura

### DISCUSSION THREAD

**Discussion by websolutions-hamburg** [vitest-dev/vitest#2950] [Q&A]
Hello,

I have a strange performance issue on a macOS Ventura 13.2.1 system. When I run Vitest (or Jest) for the first time, it takes some time to run it. It's the first run, so maybe that's okay because we have to create caches first, for example.
The second run then is very fast. And now it will be strange.
When I run the test again, without any changes to the test or the code and wait only 15 minutes, it is slow again. The test takes a very long time again, like the first run. A direct rerun is fast again.

A simple test has the following durations:

```TypeScript
import {describe, expect, test} from "vitest";

describe("Math", ()=> {
    test("Addition", ()=> {
        expect(2+3).toBe(5);
    })
})
```

**First run**: Duration  431ms (transform 33ms, setup 0ms, collect 15ms, tests 1ms)
**Second run**: Duration  176ms (transform 26ms, setup 0ms, collect 14ms, tests 1ms)

--- break (ca. 15min) ---

**Third run**: Duration  360ms (transform 27ms, setup 0ms, collect 27ms, tests 1ms)
**Fourth run**: Duration  181ms (transform 24ms, setup 0ms, collect 10ms, tests 1ms)

A more complex test has for example this durations (the more complex, the longer the run after a certain pause):

**First run**:  Duration  6.98s (transform 302ms, setup 2.67s, collect 57ms, tests 20ms)
**Second run**: Duration  926ms (transform 172ms, setup 270ms, collect 35ms, tests 19ms)

When I run `vitest --ui`, the node task hangs for one or two minutes before the browser opens and the tests run.
When I run the tests in a Docker container, every run is super fast, even the first one.

Does anyone know this slowed down behaviour? Is this a problem with Node.js, Vitest/Jest or with macOS? I have no clue how to continue here. 😢.

---

My system specs:
- macOS Ventura: `13.2.1`
- Node.js: `19.7.0`
- vitest: `0.29.2`
- @vitest/ui: `0.29.2`
- typescript: `4.9.5`

**Comment by sheremet-va** [comment]
Just a theory, because I saw similar issues for myself. Do you have a project with a lot of files? Try limiting the tests folder with `dir` option, that should help fast-glob find tests.

**Reply by websolutions-hamburg** [reply]
I used `include` to define the path to my test files. Now I have changed to `dir`, but it's the same. The first call is slow, second call is fast 😔.
I also tested it with an empty project with only one test file and the test `expect(2+3).toBe(5)`. You can see, that the node task hangs for a moment and then the Vitest starts with the tests.
On a Docker Container on macOS it is super fast.
